cpp/src/arrow/util/decimal_test.cc

```diff
@@ -1046,42 +1046,51 @@ using ToDoubleTestParam = ToRealTestParam<double>;
 template <typename Decimal, typename Real>
 void CheckDecimalToReal(const std::string& decimal_value, int32_t scale, Real expected) {
   Decimal dec(decimal_value);
-  ASSERT_EQ(dec.template ToReal<Real>(scale), expected)
-      << "Decimal value: " << decimal_value << " Scale: " << scale;
+  Real actual = dec.template ToReal<Real>(scale);
```

Comment:
I'm curious: is it possible to set the precision for gtest outputs like `std::setprecision`. The default precision won't be very helpful for small diffs.

Reply:
I wonder if that's necessary. Introducing a deliberate error seems to display the values with sufficient precision:
```
[ RUN      ] TestDecimalToRealFloat/0.Precision
/arrow/cpp/src/arrow/util/decimal_test.cc:1071: Failure
Value of: std::abs(actual - expected) <= epsilon
  Actual: false
Expected: true
Decimal value: 1.987748987892758765582589910934859345, scale: 36, expected: 1.9877489805221558, actual: 1.9877490997314453
[  FAILED  ] TestDecimalToRealFloat/0.Precision, where TypeParam = arrow::Decimal128 (0 ms)
```

Reply:
I see. It looks ok to me!